internal/charmstore/debug.go

```diff
@@ -15,8 +18,75 @@ func serveDebugInfo(http.Header, *http.Request) (interface{}, error) {
 	return appver.VersionInfo, nil
 }
 
-func newServiceDebugHandler() http.Handler {
+// GET /debug/check.
+func debugCheck(checks map[string]func() error) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		n := len(checks)
+		type result struct {
+			name string
+			err  error
+		}
+		c := make(chan result)
+		for name, check := range checks {
+			go func(n string, check func() error) {
```

Comment:
I'd tend to prefer:

     for name, check := range checks {
          name, check := name, check
          go func() {
               c <- result{name: name, err: check()}
          }()
      }

as there's then no way you can get it wrong inside
the closure because the original names are hidden.
(and more concise and arguably easier to read,
once you're familiar with the idiom)